### PR TITLE
Developer Tools Update 2020/11

### PR DESCRIPTION
**Pull request by kare**
- Rename test-rest to test-e2e
- Add recipes shell, ps and update
- Move e2e testing recipes to rest-e2e-tests/Makefile

### CODE REVIEW

File: rest-e2e-tests/Makefile [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+endef
+
+all: clean vendor test/server/wait test ## Set up and test
+
+node_modules: ## Run 'npm ci' if directory doesn't exist
+	$(call npm, ci)
+
+vendor: node_modules ## Install source code packages
+
+engine_status_check_url := http://localhost:8081/streamr-core/api/v1/users/me
```
Comment (teogeb) [inline rest-e2e-tests/Makefile:38]: Use http://localhost/api/v1/users/me

File: rest-e2e-tests/Makefile [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		if [ $$i -eq 70 ]; then
+			echo "test/server/wait: timeout while waiting for engine to come online" 1>&2
+			exit 1
+		fi
+	done
+
+.PHONY: test
+test: vendor test/e2e test/stress ## Run tests
+
+.PHONY: test/e2e
```
Comment (teogeb) [inline rest-e2e-tests/Makefile:63]: Why naming convention is different from main makefile where names are like `test-integration`?

**Review by teogeb — APPROVED** [review]

File: rest-e2e-tests/Makefile (hunk 1 above)
Comment (kare) [reply rest-e2e-tests/Makefile:38]: Changing url breaks the logic of the script. Nginx replies too quick while engine is down.

File: rest-e2e-tests/Makefile (hunk 2 above)
Comment (kare) [reply rest-e2e-tests/Makefile:63]: It's an experimental feature hidden behind the main Makefile. Let's see if those test target's get any use and if they are better the old way. Consider 'test/' as root 'folder' for all test related recipes and you can easily check what's available with `make test\t`.

**Pull request merged**